src/test/java/com/intel/chimera/stream/PositionedCryptoInputStreamTest.java

```
@@ -0,0 +1,330 @@
+      int remaining = dst.remaining();
+      final byte[] tmp = getBuf();
+      int read = 0;
+      int offset = 0;
+      while (remaining > 0) {
+        int copyLen = Math.min(Math.min(remaining, bufferSize), available());
+        if (copyLen == 0) {
+          break;
+        }
+        System.arraycopy(buf, (int)pos, tmp, 0, copyLen);
+        dst.put(tmp, offset, copyLen);
+        read += copyLen;
+        offset += copyLen;
+        pos += copyLen;
+        remaining -= copyLen;
+      }
+      return read;
+    }
+
+    @Override
+    public long skip(long n) throws IOException {
+      long k = count - pos;
+      if (n < k) {
+        k = n < 0 ? 0 : n;
+      }
+
+      pos += k;
+      return k;
+    }
+
+    @Override
+    public int read(long position, byte[] buffer, int offset, int length)
+            throws IOException {
+      if (buffer == null) {
+        throw new NullPointerException();
+      } else if (offset < 0 || length < 0 || length > buffer.length - offset) {
+        throw new IndexOutOfBoundsException();
+      }
+
+      if (position < 0 || position >= count) {
+        return -1;
+      }
+
+      long avail = count - position;
+      if (length > avail) {
+        length = (int)avail;
+      }
+      if (length <= 0) {
+        return 0;
+      }
+      System.arraycopy(buf, (int)position, buffer, offset, length);
+      return length;
+    }
+
+    @Override
+    public void readFully(long position, byte[] buffer, int offset, int length)
+            throws IOException {
+      int nread = 0;
+      while (nread < length) {
+        int nbytes = read(position+nread, buffer, offset+nread, length-nread);
+        if (nbytes < 0) {
+          throw new EOFException("End of file reached before reading fully.");
+        }
+        nread += nbytes;
+      }
+    }
+
+    @Override
+    public void seek(long position) throws IOException {
+      if (position >= 0 && position < count) {
```

1. position should be able to equal count which means the end of the stream. The current logic will never be able to seek to the end.
2. Shall we throw an exception if the position is invalid, for example negative case?